packages/editor/src/components/collab-sidebar/comment-indicator-toolbar.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * WordPress dependencies
+ */
+import { ToolbarButton } from '@wordpress/components';
+import { _x } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import {
+	privateApis as blockEditorPrivateApis,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { store as editorStore } from '../../store';
+
+const { CommentIconToolbarSlotFill } = unlock( blockEditorPrivateApis );
+
+const CommentAvatarIndicator = ( { onClick } ) => {
+	const { threadParticipants, hasUnresolved } = useSelect( ( select ) => {
+		const { getBlockAttributes, getSelectedBlockClientId } =
+			select( blockEditorStore );
+		const { getCurrentPostId } = select( editorStore );
+		const selectedBlock = getSelectedBlockClientId();
+		const selectedBlockAttributes = selectedBlock
+			? getBlockAttributes( selectedBlock )
+			: null;
+		const postId = getCurrentPostId();
+
+		// Get comment data for this block
+		const blockCommentIdValue = selectedBlockAttributes?.blockCommentId;
+		const participantsMap = new Map();
+		let isResolved = false;
+
+		if ( blockCommentIdValue && postId ) {
+			const comments = select( coreStore ).getEntityRecords(
+				'root',
+				'comment',
+				{
+					post: postId,
+					type: 'block_comment',
+					status: 'any',
+					per_page: 100,
```

Comment:
I realize that this limit is built into the REST API, however we probably need to account for posts with over 100 comments (or replies to comments).